java/README.md

```diff
@@ -1,113 +1,245 @@
-source "$HOME/.cargo/env"
+echo $JAVA_HOME
+java -version
 ```
 
-**Dependencies for MacOS**
+### Adding the client to your project
 
-Ensure that you have a minimum Java version of JDK 11 installed on your system:
-```bash
- $ echo $JAVA_HOME
-/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
+Refer to https://central.sonatype.com/artifact/software.amazon.glide/glide-for-redis.
+Once set up, you can run the basic examples.
 
-$ java -version
- openjdk version "11.0.1" 2018-10-16
- OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
- OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
-```
+Additionally, consider installing the Gradle plugin, [OS Detector](https://github.com/google/osdetector-gradle-plugin) to help you determine what classifier to use.
 
-#### Building and installation steps
-The Java client is currently a work in progress and offers no guarantees. Users should build at their own risk.
+## Classifiers
+There are 4 types of classifiers for GLIDE for Valkey which are
+``
+osx-aarch_64
+osx-x86_64
+linux-aarch_64
+linux-x86_64
+``
 
-Before starting this step, make sure you've installed all software requirements.
-1. Clone the repository:
-```bash
-VERSION=0.1.0 # You can modify this to other released version or set it to "main" to get the unstable branch
-git clone --branch ${VERSION} https://github.com/aws/glide-for-redis.git
-cd glide-for-redis
-```
-2. Initialize git submodule:
-```bash
-git submodule update --init --recursive
-```
-3. Generate protobuf files:
-```bash
-cd java/
-./gradlew :client:protobuf
+Gradle:
+- Copy the snippet and paste it in the `build.gradle` dependencies section.
+- **IMPORTANT** must include a `classifier` to specify your platform.
+```groovy
+
+osx-aarch_64
+dependencies {
+    implementation group: 'software.amazon.glide', name: 'glide-for-redis', version: '0.4.3', classifier: 'osx-aarch_64'
+}
+
+osx-x86_64
+dependencies {
+    implementation group: 'software.amazon.glide', name: 'glide-for-redis', version: '0.4.3', classifier: 'osx-x86_64'
+}
+
+linux-aarch_64
+dependencies {
+    implementation group: 'software.amazon.glide', name: 'glide-for-redis', version: '0.4.3', classifier: 'linux-aarch_64'
+}
+
+linux-x86_64
+dependencies {
+    implementation group: 'software.amazon.glide', name: 'glide-for-redis', version: '0.4.3', classifier: 'linux-x86_64'
+}
 ```
-4. Build the client library:
-```bash
-cd java/
-./gradlew :client:build
+
+Maven:
+- **IMPORTANT** must include a `classifier`. Please use this dependency block and add it to the pom.xml file.
+```xml
+
+osx-aarch_64
+<dependency>
+   <groupId>software.amazon.glide</groupId>
+   <artifactId>glide-for-redis</artifactId>
+   <classifier>osx-aarch_64</classifier>
+   <version>0.4.3</version>
+</dependency>
+
+osx-x86_64
+<dependency>
+   <groupId>software.amazon.glide</groupId>
+   <artifactId>glide-for-redis</artifactId>
+   <classifier>osx-x86_64</classifier>
+   <version><VERSION></version>
```

Comment:
Above is a specific version, and here is the `VERSION.` Shall we maintain and keep readme always to the latest version?

Reply:
IMO we should script the process of updating the version and have that script update this README.